app/dose-amigos-user-service/dose-amigos-user.service.ts

```diff
@@ -11,6 +11,9 @@ export class DoseAmigosUserService {
 
     private amigosUsersUrl: string = `${BACKEND_URL}/amigo-users`;
 
+    /* Cached currentUser so we don't have to get from server again. */
+    private currentUser: DoseAmigosUser;
```

Comment:
This service now will cache the currentUser and return it on later calls to `getCurrent()` to prevent the network call.